lib/draper/automatic_delegation.rb

```diff
@@ -6,8 +6,7 @@ module AutomaticDelegation
     def method_missing(method, *args, &block)
       return super unless delegatable?(method)
 
-      self.singleton_class.delegate method
-      send(method, *args, &block)
+      object.send(method, *args, &block)
```

Comment:
@syguer I'm not sure how to prevent the `super` call from reaching `object` because by the time we hit this `method_missing`, there is no way to tell that the method was called from `super` because it has been resolved to the method name:

![image](https://cloud.githubusercontent.com/assets/651946/24589775/081eb424-17a6-11e7-872a-ad3d5aec2258.png)

The runtime exception from the screenshot is due to me explicitly raising in `Post#bar` to get the call stack for demonstration purposes. This doesn't have anything to do with the behavior being discussed.

So as the screenshot illustrates:

1. Calling `super` within `PostDecorator#bar` tries to call `ApplicationDecorator#bar`.
2. This triggers `AutomaticDelegation#method_missing` with the name of the method (`:bar`).
3. At this point, I have no idea that `bar` was called through `super` and it sees that `Post#bar` exists and calls that.


I'm not sure how to stop this behavior. Right now, I'm leaning toward leaving the behavior as is and:

1. Documenting that `super` will try to call a parent class and if the method doesn't exist, it will try to call the method on `object`.
2. Ensuring the tests are covering this functionality
3. Merging this fix as is (since it actually fixes another issue not entirely related to `super`).

What do you think?